fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java

```diff
@@ -4788,7 +4829,21 @@ private void createTablets(String clusterName, MaterializedIndex index, ReplicaS
         Preconditions.checkArgument(replicationNum > 0);
 
         DistributionInfoType distributionInfoType = distributionInfo.getType();
-        if (distributionInfoType == DistributionInfoType.HASH) {
+        if (distributionInfoType != DistributionInfoType.HASH) {
+            throw new DdlException("Unknown distribution type: " + distributionInfoType);
+        }
+
+        if (tabletMeta.isUseStarOS()) {
+            // TODO: support colocate table
+            int bucketNum = distributionInfo.getBucketNum();
```

Comment:
I have no idea about the difference between bucketNum and tablet id, can you explain it clearly?